llama_cpp/llama_cpp.py

```diff
@@ -1,17 +1,49 @@
+import sys
+import os
 import ctypes
-
 from ctypes import c_int, c_float, c_char_p, c_void_p, c_bool, POINTER, Structure, Array, c_uint8, c_size_t
-
 import pathlib
-from itertools import chain
 
 # Load the library
-# TODO: fragile, should fix
-_base_path = pathlib.Path(__file__).parent
-(_lib_path,) = chain(
-    _base_path.glob("*.so"), _base_path.glob("*.dylib"), _base_path.glob("*.dll")
-)
-_lib = ctypes.CDLL(str(_lib_path))
+def load_shared_library(lib_base_name):
```

Comment:
Rename to `_load_shared_library` to avoid it appearing in any public APIs

Reply:
Sorry, I saw you marked this as resolved, is there a reason to not have this be `_load_shared_library`?

Reply:
@MillionthOdin16 oh I see the intention is to be able to patch the library with a custom dll at runtime correct?

Reply:
Nope, LGTM. Thank you for this!